code/modules/organs/subtypes/diona.dm

```diff
@@ -1,22 +1,19 @@
-/proc/spawn_diona_nymph(var/turf/target)
-	if(!istype(target))
-		return 0
-
+/turf/proc/spawn_diona_nymph()
 	//This is a terrible hack and I should be ashamed.
 	var/datum/seed/diona = SSplants.seeds["diona"]
 	if(!diona)
 		return 0
 
-	spawn(1) // So it has time to be thrown about by the gib() proc.
-		var/mob/living/carbon/alien/diona/D = new(target)
-		var/datum/ghosttrap/plant/P = get_ghost_trap("living plant")
-		P.request_player(D, "A diona nymph has split off from its gestalt. ")
-		spawn(60)
-			if(D)
-				if(!D.ckey || !D.client)
-					D.death()
-		return 1
+	INVOKE_ASYNC(src, .proc/create_diona_nymph)
 
+/turf/proc/create_diona_nymph()
```

Comment:
please make this a ghostspawner

Reply:
you can take a look at my apprentice pebble code to get more info

Reply:
out of scope

Reply:
don't think this was the right PR to do it in, then